File: backend/app/schemas/team.py
from datetime import datetime
from typing import Literal

from app.schemas.common import CamelModel

TeamRole = Literal["owner", "admin", "sales_rep"]


class TeamMember(CamelModel):
    id: str
    name: str
    email: str
    role: TeamRole
    avatar_src: str | None = None
    invited_at: datetime
    status: Literal["active", "invited"]


class TeamInviteInput(CamelModel):
    email: str
    role: TeamRole = "sales_rep"

File: frontend/lib/schema/team.ts
import { z } from "zod";

export const TeamRoleSchema = z.enum(["owner", "admin", "sales_rep"]);
export type TeamRole = z.infer<typeof TeamRoleSchema>;

export const TeamMemberSchema = z.object({
  id: z.string(),
  name: z.string(),
  email: z.string(),
  role: TeamRoleSchema,
  avatarSrc: z.string().nullable(),
  invitedAt: z.string(),
  status: z.enum(["active", "invited"]),
});
export type TeamMember = z.infer<typeof TeamMemberSchema>;
